spotinst/commons/common_organization_programmatic_user.go

```diff
@@ -0,0 +1,117 @@
+package commons
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/service/organization"
+	"log"
+)
+
+const (
+	OrgProgUserResourceName ResourceName = "spotinst_organization_programmatic_user"
+)
+
+var OrgProgUserResource *OrgProgUserTerraformResource
+
+type OrgProgUserTerraformResource struct {
+	GenericResource
+}
+
+type OrgProgUserWrapper struct {
+	orgProgUser *organization.ProgrammaticUser
+}
+
+func NewOrgProgUserResource(fieldsMap map[FieldName]*GenericField) *OrgProgUserTerraformResource {
+	return &OrgProgUserTerraformResource{
+		GenericResource: GenericResource{
+			resourceName: OrgProgUserResourceName,
+			fields:       NewGenericFields(fieldsMap),
+		},
+	}
+}
+
```

Comment:
let's the hierarchy same across all resources: OnCreate, OnRead, OnUpdate

Reply:
changed in same order